submariner/IPV6-OVN.md

```diff
@@ -0,0 +1,102 @@
+# **Submariner Enhancement for IPv6 Support**
+
+## **Summary**
+
+This proposal outlines the changes required in Submariner for  OVN Kubernetes
+CNI to enable IPv6 support, ensuring seamless connectivity between clusters
+using Submariner. The main proposal has the full design
+[IPv6 Datapath Enhancements](https://github.com/submariner-io/enhancements/blob/devel/submariner/IPV6-datapath.md)
+This covers only the OVN CNI part of it.
+
+---
+
+## Design Details
+
+The OVNKubernetes handler programs network policies and routes to direct traffic from
+the gateway and non-gateway nodes to direct the traffic to the remote cluster.
+At present the routes are only programmed for IPv4 for addresses. We need to enhance
+this to support IPV6 addresses as well.
+
+The handler for creating the Gateway and NonGateway routes needs to be enhanced.
+
+### GatewayRoute CRD
+
+The GatewayRoute will now populate Ipv4 and Ipv6 addresses for the next hops and
+remote CIDRs for a dual-stack environment. For Ipv6 only it uses just the IPv6
+address for these fields. Ipv4 will continue as it is.
+
+The next hop will be the interface IP of ovn-k8s-mp0 interface, which is expected
```

Comment:
When we create the GatewayRoute and NonGatewayRoute CRs, we are essentially programming OVN Openflow pipeline and will be using the GENEVE tunnels already created by OVN itself for both v4 and v6 traffic right?